### ISSUE TITLE
Pry doesn't work well  with bootsnap

### ISSUE THREAD

**Issue by aniketstiwari** [Shopify/bootsnap#327]
Whenever i place debugger for the first time it works fine. But for the second time the debugger goes into ruby internal. I check this seems to be an existing issue with byebug https://github.com/deivid-rodriguez/byebug/issues/452

I tried few things by adding disable_trace: false still the issue is reproducible  https://github.com/Shopify/bootsnap/issues/281

My bootsnap configuration

Bootsnap.setup(
    cache_dir:            'tmp/cache',          # Path to your cache
    development_mode:     'development', # Current working environment, e.g. RACK_ENV, RAILS_ENV, etc
    load_path_cache:      true,                 # Optimize the LOAD_PATH with a cache
    autoload_paths_cache: true,                 # Optimize ActiveSupport autoloads with cache
    compile_cache_iseq:   true,                 # Compile Ruby code into ISeq cache, breaks coverage reporting.
    compile_cache_yaml:   true,                  # Compile YAML into a cache
    disable_trace: false
  )

Rails - 5.2.4.4
ruby - 2.5.3
Bootsnap - 1.4.1
pry - 0.12.2
pry-rails 0.3.9

**Comment by casperisfine** [comment]
Pretty sure it is the same root cause than https://github.com/Shopify/bootsnap/issues/244

TL;DR; Ruby 2.5.x doesn't fire tracepoint events when code is loaded from cache. It's better to set `compile_cache_iseq: false` when running with Ruby 2.5.

I'll be adding a warning for this in the next version.

Closing.
